<a href="https://colab.research.google.com/github/RheaMimiCarillo/skillspire-ai-ml/blob/main/ADA_Lab_8.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Rhea Mimi Carillo

August 2026

#Part 1: CONCEPTUAL QUESTIONS

**1. What differs data science from data analytics?**

> Data Science is a larger umbrella that encapsulates Data Analytics.


**2. Between mean and median, what do you think is more sensitive to outliers? Why?**



> The mean is more sensitive to outliers, because it uses the sum of values. The mean can be skewed by extremely high or low values. We prefer to use the median when there are major outliers skewing the mean.

**3. Is correlation also causation? Why or why not?**

> Correlation is not causation.
> > **Here's a *Faulty* Example to illustrate what I mean:** The mean temperature lowers from Independence Day to Christmas in the the northern hemisphere. The temperature also climbs from Christmas to the following Independence Day. Temperature is strongly correlated with the occurance of Independence Day & Christmas.
> > > **Faulty conclusion:** Thus, the Christmas holiday causes the local temperature to lower and Independence Day causes the temperature to rise.
> > In reality, we know that holidays do not cause the weather to change.

#Part 2: Correlation Coefficients

Interpret the following in terms of mean, std, and correlation coefficient:

In [1]:
import numpy as np

# Manually create a dataset with 15 observations
coffee_consumption = [2, 4, 1, 5, 3, 1, 3, 2, 4, 5, 1, 4, 3, 2, 5]
productivity_scores = [60, 75, 50, 80, 70, 55, 65, 60, 72, 85, 48, 78, 68, 58, 90]

In [26]:
# Coffee vs Productivity

import pandas as pd

print(f"Coffee Mean: {np.mean(coffee_consumption)}")
print(f"Productivity Mean: {np.mean(productivity_scores)}")

print(f"\nCoffee Median: {np.median(coffee_consumption)}")
print(f"Productivity Median: {np.median(productivity_scores)}")

print(f"\nCoffee Standard Deviation: {np.std(coffee_consumption).round(2)}")
print(f"Productivity Standard Deviation: {np.std(productivity_scores).round(2)}")


print(f"\nCorrelation Coefficient:\n{np.corrcoef(coffee_consumption, productivity_scores).round(2)}")

coffee_productivity = {
    "Coffee Consumption": coffee_consumption,
    "Productivity Score": productivity_scores
}


df = pd.DataFrame(coffee_productivity)

df

Coffee Mean: 3.0
Productivity Mean: 67.6

Coffee Median: 3.0
Productivity Median: 68.0

Coffee Standard Deviation: 1.41
Productivity Standard Deviation: 12.15

Correlation Coefficient:
[[1.   0.97]
 [0.97 1.  ]]


,Coffee Consumption,Productivity Score
0,2,60
1,4,75
2,1,50
3,5,80
4,3,70
5,1,55
6,3,65
7,2,60
8,4,72
9,5,85


### Coffee vs. Productivity Interpretation

Mean Coffees consumed is **3.0** with a standard deviation of **1.41**. This means that the average coffee's consumed is 3.0 coffees, give or take 1.41 coffees.

Mean Productivity score is **67.6** with a standard deviation of **12.15**. This means that participant score centered around 67.6, give or take 12.15 points.

The correlation coefficient of +0.97 indicates a **strong correlation** between Coffee Consumption & Productivity Scores

In [22]:
import numpy as np

# Manually create a dataset with 15 observations
hours_of_tv = [2, 3, 2, 4, 5, 1, 3, 2, 4, 5, 1, 4, 3, 2, 5]
grades = [78, 80, 75, 82, 85, 70, 79, 77, 81, 88, 69, 84, 79, 76, 87]


In [25]:
# Hours of TV vs Grades

print(f"TV Mean: {np.mean(hours_of_tv).round(2)}")
print(f"Grades Mean: {np.mean(grades).round(2)}")

print(f"\nTV Median: {np.median(hours_of_tv)}")
print(f"Grades Median: {np.median(grades)}")

print(f"\nTV Standard Deviation: {np.std(hours_of_tv).round(2)}")
print(f"Grades Standard Deviation: {np.std(grades).round(2)}")


print(f"\nCorrelation Coefficient:\n{np.corrcoef(hours_of_tv, grades).round(2)}")

tv_vs_grades = {
    "Hours of TV": hours_of_tv,
    "Grades": grades
}


df1 = pd.DataFrame(tv_vs_grades)

df1

TV Mean: 3.07
Grades Mean: 79.33

TV Median: 3.0
Grades Median: 79.0

TV Standard Deviation: 1.34
Grades Standard Deviation: 5.35

Correlation Coefficient:
[[1.   0.96]
 [0.96 1.  ]]


,Hours of TV,Grades
0,2,78
1,3,80
2,2,75
3,4,82
4,5,85
5,1,70
6,3,79
7,2,77
8,4,81
9,5,88


### TV vs. Grades Interpretation

The mean Hours of TV watched is 3.07 with a standard deviation of 1.34.

The mean grade score is 79.33 with a standard deviation of 5.35.

The correlation coefficient of +0.96 indicates a **strong correlation** between Hours of TV & Grades.

In [28]:
import numpy as np

# Manually create a dataset with 15 observations
x = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
y = [2, 4, 6, 8, 45, 12, 14, 45, 18, 20, 22, 24, 46, 28, 45]


In [30]:
# Hours of X vs Y

print(f"X Mean: {np.mean(x).round(2)}")
print(f"Y Mean: {np.mean(y).round(2)}")

print(f"\nX: {np.median(x)}")
print(f"Y: {np.median(y)}")

print(f"\nX: {np.std(x).round(2)}")
print(f"Y: {np.std(y).round(2)}")


print(f"\nCorrelation Coefficient:\n{np.corrcoef(x, y).round(2)}")

x_vs_y = {
    "X": x,
    "Y": y
}


df2 = pd.DataFrame(x_vs_y)

df2

X Mean: 8.0
Y Mean: 22.6

X: 8.0
Y: 20.0

X: 4.32
Y: 15.4

Correlation Coefficient:
[[1.   0.66]
 [0.66 1.  ]]


,X,Y
0,1,2
1,2,4
2,3,6
3,4,8
4,5,45
5,6,12
6,7,14
7,8,45
8,9,18
9,10,20


### X vs. Y Interpretation

X mean is 8.0 with a standard deviation of 4.32.

Y mean is 20.0 with a standard deviation of 15.4.

The correlation coefficient of +0.66 indicates a **moderately high correlation** between X & Y.


X Mean: 8.0
Y Mean: 22.6

X: 8.0
Y: 20.0

X: 4.32
Y: 15.4

Correlation Coefficient:
[[1.   0.66]
 [0.66 1.  ]]

#Part 3: Probability

**1. Find the probability of getting a numbered card when a card is drawn from the pack of 52 cards.**

> $P(\text{Drawing a numbered card from a standard 52-card deck})$  
> $= P(\text{Drawing 2, 3, 4, 5, 6, 7, 8, 9, 10})$  
> $= P(\frac{36}{52})$  
> $= 0.69230769$

There are 36 numbered cards in a standard 52-card deck, so the probability for pulling a numbered card from a freshly-shuffled deck is $\frac{36}{52}$, or **0.69230769**.

**2. You have a bag containing 8 red balls, 5 green balls, and 3 blue balls. If you randomly draw two balls without replacement, what is the probability that both balls are green?**

> $P(\text{Picking two green balls in a row without replacement})$  
> $P(\frac{5}{8 + 5 + 3}) \times P(\frac{4}{8 + 4 + 3})$  
> $= P(\frac{5}{16}) \times P(\frac{4}{15})$  
> $= 0.3125 \times 0.26666667$  
> $= 0.08333333$

**3. You roll a six-sided die. What is the probability of rolling an even number or a number greater than 4?**

> $P(\text{Even}) \cup P(\text{5 or 6}) = P(\{2, 4, 5, 6\})$  
> $= \frac{4}{6} = \frac{2}{3}$  
> $\approx 0.667$

6 is both even and greater than 4, so it's only counted once for this set.